From dd6f7b3336df839a8818e16a1173bc67912e519f Mon Sep 17 00:00:00 2001
From: Will Thompson <will@willthompson.co.uk>
Date: Sat, 17 Feb 2024 18:16:08 +0000
Subject: [PATCH 1/3] appdata: Replace named display_length with constant
 values

The named values for <display_length> were removed in Appstream 1.0,
released recently.  In their place, use integer values.

For the <requires> block, attempt to encode the same reasonableness
restrictions as are enforced on the windowsize setting. The code
requires the height to be at least 480px and the width at least 500px,
but there is no way to specify width versus height, just shortest versus
longest edge, so I just require that the shortest edge is at least
500px.

For the <recommends> block, 768px is the suggested number in the
appstream specification for laptops and tablets.

https://github.com/ximion/appstream/commit/005c5f104d45953cd25f842b70dcd3d32466333b
https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-relations-display_length
---
 src/org.tuxpaint.Tuxpaint.appdata.xml.in | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/org.tuxpaint.Tuxpaint.appdata.xml.in b/src/org.tuxpaint.Tuxpaint.appdata.xml.in
index 6d27a9ab..c1f73175 100644
--- a/src/org.tuxpaint.Tuxpaint.appdata.xml.in
+++ b/src/org.tuxpaint.Tuxpaint.appdata.xml.in
@@ -168,10 +168,11 @@
     <control>keyboard</control>
     <control>pointing</control>
     <control>touch</control>
-    <display_length compare="ge">medium</display_length>
+    <display_length compare="ge">768</display_length>
   </recommends>
   <requires>
-    <display_length compare="ge">small</display_length>
+    <display_length compare="ge">500</display_length>
+    <display_length compare="le">32000</display_length>
   </requires>
   <translation type="gettext">tuxpaint</translation>
   <launchable type="desktop-id">tuxpaint.desktop</launchable>
-- 
2.43.0

